patches/tModLoader/Terraria/ModLoader/MenuLoader.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.Audio;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader.Default;
+using Terraria.UI.Chat;
+
+namespace Terraria.ModLoader
+{
+	internal static class MenuLoader
+	{
+		internal static ModMenu currentModMenu;
+
+		private static readonly List<ModMenu> moddedMenus = new List<ModMenu>();
+
+		private static List<ModMenu> lastAvailableMenus = new List<ModMenu>();
+
+		internal static List<ModMenu> AvailableMenus => moddedMenus.Where(m => m.IsAvailable).ToList();
+
+		internal static string lastUsedModMenuName = $"ModLoader/{nameof(MenutML)}";
+
+		internal static void Add(ModMenu modMenu) {
+			moddedMenus.Add(modMenu);
+		}
+
+		private static void OffsetModMenu(int offset) {
+			var menus = AvailableMenus;
+
+			currentModMenu = menus[Utils.Repeat(menus.IndexOf(currentModMenu) + offset, menus.Count)];
+		}
+
+		internal static void GotoSavedModMenu() {
+			if (lastUsedModMenuName == nameof(MenuOldVanilla)) {
+				Main.instance.playOldTile = true; // If the previous menu was the 1.3.5.3 one, automatically reactivate it.
+			}
+			List<ModMenu> menus = AvailableMenus;
+			int index = menus.FindIndex(m => GetModMenuName(m) == lastUsedModMenuName);
+			if (index == -1) {
+				index = 0;
+			}
+			currentModMenu = menus[index];
+			currentModMenu.SelectionInit();
+			lastUsedModMenuName = GetModMenuName(currentModMenu);
```

Comment:
the 2 lines above look redundant, should be handled by the start of `UpdateAndDrawModMenu`

Reply:
those exist for when the game opens to a modded menu, because `lastUsedModMenuName` will be equal to the current ModMenu, so SelectionInit wouldn't fire. I can remove the assignment to lastUsedModMenuName but probably not the SelectionInit call